transport-msg.tex

```diff
@@ -940,33 +971,56 @@ \subsubsection{Overview}
 & 20 & 8 & Driver address \\
 & 28 & 8 & Device address \\
 \hline
-Answer & 0 & 4 & Virtqueue index \\
-& 4 & 4 & Reserved (Must Be Zero - MBZ) \\
-& 8 & 4 & Current virtqueue size \\
-& 12 & 8 & Descriptor address \\
-& 20 & 8 & Driver address \\
-& 28 & 8 & Device address \\
+Answer & 0 & 0 & no extra data \\
 \hline
 \end{tabular}
 
 \msgdef{RESET_VQUEUE}
 
-This message is sent by the virtio-msg transport frontend and requires a
-response from the backend.
+This message is sent by the virtio-msg transport driver and requires a
+response from the device.
+
+The driver \emph{SHOULD NOT} send this message unless the VIRTIO_F_RING_RESET
+feature has been negotiated.
 
 \begin{tabular}{|l|l|l|l|}
 \hline
 Type & Offset & Size (bytes) & Content \\
 \hline \hline
 Request & 0 & 4 & Virtqueue index \\
 \hline
-Answer & 0 & 0 & None \\
+Answer & 0 & 0 & no extra data \\
+\hline
+\end{tabular}
+
+\msgdef{EVENT_CONFIG0}
```

Comment:
Could we try to find an other name than EVENT_CONFIG0 which is very near to EVENT_CONFIG ?
Maybe event CONFIG_EMPTY and CONFIG_INFO ?

Reply:
EVENT_CONFIG and EVENT_CONFIG0 are the same message with different amounts of data.  I would prefer we just use msg length.  HOWEVER we can use EVENT_CONFIG_EMPTY if you don't like msg length and don't like EVENT_CONFIG0

Reply:
Using the length is a possibility I do not like because it makes some parts of the message optionals (it is ok when you have a variable length table at the end). And here we must not only remove the data but also the status.
Could we somehow define that a status of 0 means "invalid" (or an other value) so that the transport knows that the status must be retrieved and we would not need CONFIG0 anymore ?

Reply:
Status of 0 is a valid device state so I don't think we should do that.
Our status is 32 bits as is virtio-mmio's field.
The common spec only used values <0xFF so we could define 0x8000_0000 or 0xFFFF_FFFF as invalid but the common spec could invent things for the other bits.

We could add yet another field for valid but I would prefer a different message than that.

I will change the name of EVENT_CONFIG0 to EVENT_CONFIG_EMPTY for now and we can discuss at the sprint.